src/vision/status_quo/niklaus_lost_his_context.md

```diff
@@ -0,0 +1,99 @@
+# 😱 Status quo stories: Niklaus lost his context!
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Niklaus heard about a project to reimplement a deprecated browser plugin using Rust and WASM. This old technology had the ability to load resources over HTTP; so it makes sense to try and implement that functionality using the Fetch API. He looks up the documentation of `web_sys` and realizes that he needs to take a thing called a `Future` out of the call to `fetch`, do some things to it in Rust, and then send the future off to `spawn_local` as such:
+
+```use wasm_bindgen_futures::spawn_local;
+use web_sys::window;
+
+fn load_image(src: &url) {
+    spawn_local(async move {
+        let url = src.to_string();
+        let request = //request construction in Rust is actually pretty verbose
+                      //just pretend it's here
+        window().unwrap().fetch_with_request(&request).await;
+        log::error!("It worked");
+    });
+}
+```
+
+Niklaus adds calls to `load_image` where appropriate, sees the message pop up in the console, and figures it's time to now actually do something to that loaded image. At this point, it's important to note that many parts of this project pass around "contexts" - bundles of borrowed state that various parts of the reimplementation are free to alter. (This is necessary because this plugin had multiple scripting runtimes in it.) In synchronous code, this was perfectly fine. However, when we start integrating the function into the player...
+
+```use crate::{Index, Context}
+use wasm_bindgen_futures::spawn_local;
+use web_sys::window;
+use crate::parse_png;
+
+fn load_image(src: &url, inside_of: Index<Node>, context: &mut Context<'_>) {
```

Comment:
I'd like to know a bit more about where this `context` comes from. =) I guess it is under Alan's control. I'm trying to figure out if this overall plugin is 'conceptually single threaded" or what -- could `fn load_image` be an `async fn`? Presumably not.

It sounds like what Alan probably wants is this `context` to be "ambiently available" to the futures, and to rely on the fact that there should be only one running at a time so that it can be `&mut`? Is it kind of way to setup context and for things to communicate?

Reply:
You are correct, it is multithreaded - half because we're in WASM and WASM multithreading is a mess right now, and half because the actual thing we're reimplementing also lived in a browser mainthread.

You're right about the overall intent, too. In fact, I probably should rename it to `World` to avoid confusion with executor contexts.